File: src/main/kotlin/inflearn/java/ch02_array/_2_9_격자판_최대합.java
package inflearn.java.ch02_array;

import java.util.Scanner;
import java.util.StringTokenizer;

public class _2_9_격자판_최대합 {
    public static int solution(int[][] ints) {
        int answer = Integer.MIN_VALUE;
        int sum1, sum2;
        int n = ints.length;
        for (int i = 0; i < n; i++) {
            sum1 = sum2 = 0;
            for (int j = 0; j < n; j++) {
                sum1 += ints[i][j];
                sum2 += ints[j][i];
            }
            answer = Math.max(answer, sum1);
            answer = Math.max(answer, sum2);
        }
        sum1 = sum2 = 0;
        for (int i = 0; i < n; i++) {
            sum1 += ints[i][i];
            sum2 += ints[i][n - 1 - i];
        }
        answer = Math.max(answer, sum1);
        answer = Math.max(answer, sum2);

        return answer;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = Integer.parseInt(sc.nextLine());
        int[][] input = new int[n][n];
        for (int i = 0; i < n; i++) {
            StringTokenizer st = new StringTokenizer(sc.nextLine());
            for (int j = 0; j < n; j++) {
                input[i][j] = Integer.parseInt(st.nextToken());
            }
        }

        System.out.println(solution(input));
    }
}

File: src/main/kotlin/inflearn/2_9 격자판 최대합.kt
package inflearn

import java.util.*

object `2_9 격자판 최대합` {
    fun solution(ints: Array<IntArray>): Int {
        var answer = Int.MIN_VALUE
        var sum1: Int
        var sum2: Int
        val n = ints.size
        for (i in 0 until n) {
            sum2 = 0
            sum1 = sum2
            for (j in 0 until n) {
                sum1 += ints[i][j]
                sum2 += ints[j][i]
            }
            answer = answer.coerceAtLeast(sum1)
            answer = answer.coerceAtLeast(sum2)
        }
        sum2 = 0
        sum1 = sum2
        for (i in 0 until n) {
            sum1 += ints[i][i]
            sum2 += ints[i][n - 1 - i]
        }
        answer = answer.coerceAtLeast(sum1)
        answer = answer.coerceAtLeast(sum2)
        return answer
    }

    @JvmStatic
    fun main(args: Array<String>) {
        val sc = Scanner(System.`in`)
        val n = sc.nextLine().toInt()
        val input = Array(n) { IntArray(n) }
        for (i in 0 until n) {
            val st = StringTokenizer(sc.nextLine())
            for (j in 0 until n) {
                input[i][j] = st.nextToken().toInt()
            }
        }
        println(solution(input))
    }
}